angular/src/app/metrics/metrics.component.ts

```diff
@@ -113,17 +118,18 @@ export class MetricsComponent implements OnInit {
         // Expend the data tree to level one
         this.yAxisLabel = "";
 
-        if(this.inBrowser){
-            this.route.params.subscribe(queryParams => {
-                this.ediid = queryParams.id;
-                this.pdrHomeUrl = this.lps + this.ediid;
-                // Get dataset title
-                this.searchService.searchById(this.ediid, true).subscribe(md => {
-                    if(md) {
-                        this.record = md as NerdmRes;
-                        this.datasetTitle = md['title'];
-                        this.pdrid = md['@id'];
-        
+        this.route.params.subscribe(queryParams => {
+            this.ediid = queryParams.id;
+            this.pdrHomeUrl = this.lps + this.ediid;
+            // Get dataset
+            this.mdserv.getMetadata(this.ediid).subscribe(
+                (md) => {
+                    // successful metadata request
+                    this.record = md as NerdmRes;
+                    this.datasetTitle = md['title'];
+                    this.pdrid = md['@id'];
+
+                    if (this.inBrowser) {
                         this.createNewDataHierarchy();
```

Comment:
I added this `inBrowser` guard to skip retrieving the metrics data when on the server-side.